### ISSUE TITLE
Bind events via observer

### ISSUE THREAD

**Issue by 4n70w4** [mvanduijker/laravel-transactional-model-events#2]
Hi! Thanks for package!

It will be very useful to bind events through the observer like this:

```php
Something::observe(SomethingObserver::class);
```

```php

class SomethingObserver{

    /**
     * @param Something $something
     */
    public function afterCommitSaved(Something $something): void {
        // something handler
    }
```

**Comment by mvanduijker** [comment]
That would be cool! Do you have any idea to implement this? I took a short look but couldn't find a quick way to add observers in a trait boot method.

**Comment by mvanduijker** [comment]
We could probably get this working with trait initializers. https://github.com/laravel/framework/pull/24467 I am pretty busy coming week. Hope to add this after next week.

**Comment by mvanduijker** [comment]
@4n70w4 I added support for observers in master. Could you check it out if it works as expected?

**Comment by 4n70w4** [comment]
@mvanduijker thanks! I was able to check it within 24 hours.

Offtop:
I also tried another package, but found a fatal error:
https://github.com/fntneves/laravel-transactional-events/issues/27

Do such chains work fine in this package?

**Comment by mvanduijker** [comment]
@4n70w4 cool!

Never tried it, but don't see why it shouldn't work.

**Comment by 4n70w4** [comment]
@mvanduijker It seems to work!
It would be nice to release the new version through the packagist.
Then I will look in the prod.

**Comment by mvanduijker** [comment]
@4n70w4 I released version 2.0.0 with support for observers. Thanks for requesting this improvement.
